designs/disruption-controls.md

```diff
@@ -0,0 +1,118 @@
+# Disruption Controls - (in v1beta1+)
+
+## Motivation
+Users have been asking for more control around the speed of Consolidation and the ability to restrict disruptions to their nodes. The demand for this is reflected by the +1s in [#1738](https://github.com/aws/karpenter/issues/1738). This is increasingly important as clusters can run at a highly varied scales, where some users have reported scale-down behaving too slow at higher scales, or too quick at smaller scales. Additionally, users that want to control when nodes can be disrupted must add PDBs to block node disruptions, add `do-not-evict` annotations on pods for sensitive workloads, or add `do-not-consolidate` annotations on nodes that shouldn't be consolidated. Natively integrating a better way to ensure nodes aren't disrupted will ease the burden that users have to take on with disruption.
+
+These controls will include a new `Disruption` block in the `v1beta1` `NodePool` that will contain all scale-down behavioral fields. Initially, this will include `Budgets`, which define a (1) parallelism of how many nodes can be deprovisioned at a time and a (2) cron schedule that determines when the parallelism applies, and a `ConsolidateAfter` field that will allow users to affect the speed at which Karpenter will scale down underutilized nodes. Future scale-down behavioral fields should be colocated with these fields.
+
+## Proposed Spec
+
+```yaml
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: default
+spec: # This is not a complete NodePool Spec.
+  disruption:
+    consolidationPolicy: WhenUnderutilized || WhenEmpty
+    consolidateAfter: 10m || Never # metav1.Duration
+    expireAfter: 10m || Never # Equivalent to v1alpha5 TTLSecondsUntilExpired
+    budgets:
+    # On Weekdays during business hours, don't do any deprovisioning.
+    - crontab: "0 9 * * mon-fri"
```

Comment:
kubernetes cronjobs has added support for time zones for CronJobs. I think you should consider the same here, to adjust the business hours based on daylight savings time.

If the same format is accepted as for CronJobs, I think it would be good to use the same names (`schedule` and `timeZone`)

https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#time-zones

Reply:
Yep! In the go struct spec, It's called out that     `// Crontab uses the same syntax as a Cronjob`. But I can definitely update it so that's more clear.